### PR TITLE
Adds JSON and YAML output formatters for Org Search

### PR DESCRIPTION
**Pull request by kpitzen**
# Description

Adds the ability to format search output in JSON or YAML formats.  This is useful for larger-scale querying and processing of the data post-hoc.

Fixes https://github.com/pulumi/pulumi.ai/issues/126

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [ ] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: pkg/cmd/pulumi/org_search.go [hunk 1]
```diff
@@ -93,7 +125,14 @@ func (cmd *searchCmd) Run(ctx context.Context, args []string) error {
 	if err != nil {
 		return err
 	}
-	err = cmd.RenderTable(res.Resources)
+	switch cmd.outputFormat {
+	case outputFormatJSON:
+		err = cmd.RenderJSON(res.Resources)
```
Comment (Frassle) [inline pkg/cmd/pulumi/org_search.go:130]: Feels like "Render" ought to be a method on "outputFormat" itself.

File: pkg/cmd/pulumi/org_search.go [hunk 2]
```diff
@@ -31,11 +31,43 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 	"github.com/spf13/cobra"
 	auto_table "go.pennock.tech/tabular/auto"
+	"go.uber.org/multierr"
```
[the remark this answers is not in the document]
Comment (kpitzen) [reply pkg/cmd/pulumi/org_search.go:34]: I was certain I'd removed this, but it's possible copilot/some other VSCode extension got a little excited about adding dependencies here. I'll tidy this up.

File: pkg/cmd/pulumi/org_search.go (hunk 1 above)
Comment (kpitzen) [reply pkg/cmd/pulumi/org_search.go:130]: That's nice! I like that.

File: pkg/cmd/pulumi/org_search.go [hunk 3]
```diff
@@ -31,11 +31,43 @@ import (
...
+	outputFormatJSON  outputFormat = "json"
+	outputFormatYAML  outputFormat = "yaml"
+)
+
+// String is used both by fmt.Print and by Cobra in help text
+func (e *outputFormat) String() string {
+	return string(*e)
+}
+
+// Set must have pointer receiver so it doesn't change the value of a copy
```
[the remark this answers is not in the document]
Comment (kpitzen) [reply pkg/cmd/pulumi/org_search.go:51]: I'm not entirely sure what you mean here - the `pflag.Value` interface for cobra requires the signatures which are there for these methods. Is there something I can do to "get around" this? For lack of a better term.

[the remark this answers is not in the document]
Comment (Frassle) [reply pkg/cmd/pulumi/org_search.go:51]: If this is for interoping with pflag id comment it as such, and yeh probably leave it as is. I assume pflag gives a nice error message from Set if you try and pass a bad flag?

File: pkg/cmd/pulumi/org_search.go [hunk 4]
```diff
@@ -21,7 +21,7 @@ import (
 	"io"
 	"os"
 
-	"github.com/hashicorp/go-multierror"
+	"github.com/goccy/go-json"
```
Comment (justinvp) [inline pkg/cmd/pulumi/org_search.go:24]: Why is this necessary over the regular `import "encoding/json"`?

Comment (kpitzen) [reply pkg/cmd/pulumi/org_search.go:24]: I expect it's very much not, and it likely another copilot hallucination. Let me fix this up.

File: pkg/cmd/pulumi/org_search.go [hunk 5]
```diff
@@ -129,6 +192,14 @@ func newSearchCmd() *cobra.Command {
 			"Must be formatted like: -q key1=value1 -q key2=value2. "+
 			"Alternately, each parameter provided here can be in raw Pulumi query syntax form.",
 	)
+	cmd.PersistentFlags().VarP(
+		&scmd.outputFormat, "output", "o",
+		"Output format. Supported formats are 'table', 'json', 'csv', and 'yaml'.",
+	)
```
Comment (justinvp) [inline pkg/cmd/pulumi/org_search.go:198]: Not blocking for this PR, but as a follow-up, it might be nice to add support for `--json`, which, if selected would choose JSON as the output format (and would error if you specified `--output` with some other value). This would align with many other CLI commands that support `--json`.

**Review by justinvp — APPROVED** [review]
LGTM

File: pkg/cmd/pulumi/org_search.go (hunk 5 above)
Comment (kpitzen) [reply pkg/cmd/pulumi/org_search.go:198]: Ah! Yes, I like that as well. Would you mind opening an issue on `pulumi/pulumi.ai`? Once this is ready to release, I'd love to come back and address these usability issues as well.
